os/mm/mm_heap/mm_manage_allocfail.c

```diff
@@ -91,7 +110,23 @@ void mm_manage_alloc_fail(struct mm_heap_s *heap, int startidx, int endidx, size
 	}
 #endif
 
+}
+
+#ifdef CONFIG_APP_BINARY_SEPARATION
+void mm_ioctl_alloc_fail(size_t size)
```

Comment:
mminfo is under CONFIG_MMINFO so that this will not work with disabled config. But this function must be work when CONFIG_MM_ASSERT_ON_FAIL is enabled. Please make it work always with CONFIG_MM_ASSERT_ON_FAIL enabled.